networkx/algorithms/approximation/traveling_salesman.py

```diff
@@ -509,11 +509,17 @@ def direction_of_ascent(d=None):
             min_k_d = None
             for arborescence in minimum_1_arborescences:
                 weighted_cost = 0
+                v_k_0 = 0
                 for n in G:
                     weighted_cost += d[n] * (arborescence.degree(n) - 2)
+                    if arborescence.degree(n) - 2 == 0:
+                        v_k_0 += 1
                 if weighted_cost < min_k_d_weight:
                     min_k_d_weight = weighted_cost
                     min_k_d = arborescence
+                if v_k_0 == len(G):
+                    min_k_d_weight = weighted_cost
+                    min_k_d = arborescence
```

Comment:
This code sets the `min_k_d` to the current arborescence whenever all the nodes in the arborescence have degree 2.  But that might overwrite a result with a lower weight arborescence... Am I missing something?

Reply:
So I did comment this into the code, but may have not pushed until just now. Basically, if we are at the optimal point of the function and have just called `direction_of_ascent()`, all of the arborescences had a weighted cost of zero, so it code will almost pick the first one it encounters, even if the solution we are looking for is among that set. This will result it in returning the wrong arborescence later. Honestly, I should probably break out of that loop entirely if I see a valid solution.

Reply:
As for the unrelated changes, I had noticed that the precommit hook would reformat the files but then I would have to create a second commit to include those changes so I decided to install black and run it myself. I would guess that the versions are different as it did affect more code that I was expecting it to do. 

Because I was the one how ran black, PyCharm should be fine with rolling back those changes. I will have to be more careful of where I run black in the future, but you can definitely back out of the proposed changes.